test/Ocelot.AcceptanceTests/ServiceDiscoveryTests.cs

```diff
@@ -447,19 +448,119 @@ public void should_handle_request_to_poll_consul_for_downstream_service_and_make
+            var downstreamServiceUrlEU = $"http://localhost:{servicePortEU}";
+            var upstreamHostUS = "us-shop";
+            var upstreamHostEU = "eu-shop";
+            var publicUrlUS = $"http://{upstreamHostUS}";
+            var publicUrlEU = $"http://{upstreamHostEU}";
+            var responseBodyUS = "Phone chargers with US plug";
+            var responseBodyEU = "Phone chargers with EU plug";
+            var fakeConsulServiceDiscoveryUrl = $"http://localhost:{consulPort}";
+            var serviceEntryUS = new ServiceEntry
+            {
+                Service = new AgentService
+                {
+                    Service = serviceNameUS,
+                    Address = "localhost",
+                    Port = servicePortUS,
+                    ID = Guid.NewGuid().ToString(),
+                    Tags = ["US"],
+                },
+            };
+            var serviceEntryEU = new ServiceEntry
+            {
+                Service = new AgentService
+                {
+                    Service = serviceNameEU,
+                    Address = "localhost",
+                    Port = servicePortEU,
+                    ID = Guid.NewGuid().ToString(),
+                    Tags = ["EU"],
+                },
+            };
+
+            var configuration = new FileConfiguration
+            {
+                Routes =
+                [
+                    new()
+                    {
+                        DownstreamPathTemplate = "/",
+                        DownstreamScheme = "http",
+                        UpstreamPathTemplate = "/",
+                        UpstreamHttpMethod = ["Get"],
+                        UpstreamHost = upstreamHostUS,
+                        ServiceName = serviceNameUS,
+                        LoadBalancerOptions = new() { Type = loadBalancerType },
+                    },
+                    new()
+                    {
+                        DownstreamPathTemplate = "/",
+                        DownstreamScheme = "http",
+                        UpstreamPathTemplate = "/",
+                        UpstreamHttpMethod = ["Get"],
+                        UpstreamHost = upstreamHostEU,
+                        ServiceName = serviceNameEU,
+                        LoadBalancerOptions = new() { Type = loadBalancerType },
+                    },
+                ],
+                GlobalConfiguration = new()
+                {
+                    ServiceDiscoveryProvider = new()
+                    {
+                        Scheme = "http",
+                        Host = "localhost",
+                        Port = consulPort,
+                    },
+                },
+            };
+
+            this.Given(x => x.GivenThereIsAServiceRunningOn(downstreamServiceUrlUS, "/", 200, responseBodyUS))
+                .And(x => x.GivenThereIsAServiceRunningOn(downstreamServiceUrlEU, "/", 200, responseBodyEU))

```

Comment:
I think it is wrong!
Pay attention to other tests which use 2 services.
Valid setup should be:
```suggestion
            this.Given(x => x.GivenProductServiceOneIsRunning(downstreamServiceUrlUS, "/", 200, responseBodyUS))
                .And(x => x.GivenProductServiceTwoIsRunning(downstreamServiceUrlEU, "/", 200, responseBodyEU))
```

In current `GivenThereIsAServiceRunningOn` helper you use shared private variable `_downstreamPath` deciding on status OK. But I guess it is wrong!